ros/api/v1/hosts.py

```diff
@@ -27,17 +30,15 @@
 
 class IsROSConfiguredApi(Resource):
     def get(self):
-        ident = identity(request)['identity']
-        account_query = db.session.query(RhAccount.id).filter(
-            RhAccount.account == ident['account_number']).subquery()
-        system_count = db.session.query(System.id)\
-            .filter(System.account_id.in_(account_query)).count()
-        systems_with_suggestions = db.session.query(System.id)\
-            .filter(System.account_id.in_(account_query))\
-            .filter(System.number_of_recommendations > 0).count()
-        systems_waiting_for_data = db.session.query(System.id)\
-            .filter(System.account_id.in_(account_query))\
-            .filter(System.state == 'Waiting for data').count()
+        query = (
+            db.session.query(System.id)
+            .join(PerformanceProfile, PerformanceProfile.system_id == System.id)
+            .join(RhAccount, RhAccount.id == System.account_id)
+            .distinct()
+        )
+        system_count = query.count()
```

Comment:
@apuntamb  - It should only show systems count specific to current user's account.With your changes it will show all systems count to every user regardless of her/his account.

Reply:
Yeah, missed that. Made certain changes for the same.